tests/FacebookVideoUploaderTest.php

```diff
@@ -0,0 +1,108 @@
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\Tests;
+
+use Facebook\Facebook;
+use Facebook\FileUpload\FacebookVideoUploader;
+use Facebook\Exceptions\FacebookSDKException;
+
+class FacebookVideoUploaderTest extends \PHPUnit_Framework_TestCase
+{
+    protected $testFacebook;
+    protected $testUserId;
+    protected $testUserToken;
+
+    public function setUp()
+    {
+        if (!file_exists(__DIR__ . '/FacebookTestCredentials.php')) {
+            throw new FacebookSDKException(
+                'You must create a FacebookTestCredentials.php file from FacebookTestCredentials.php.dist'
+            );
+        }
+
+        if (!strlen(FacebookTestCredentials::$appId) ||
+            !strlen(FacebookTestCredentials::$appSecret)
+        ) {
+            throw new FacebookSDKException(
+                'You must fill out FacebookTestCredentials.php'
+            );
+        }
+
+        $this->testFacebook = new Facebook([
+                'app_id' => FacebookTestCredentials::$appId,
+                'app_secret' => FacebookTestCredentials::$appSecret,
+            ]
+        );
+
+        //setup test user
+        $testUserPath = '/' . FacebookTestCredentials::$appId . '/accounts/test-users';
+
+        $params = [
+            'installed' => true,
+            'name' => 'Foo Phpunit User',
+            'locale' => 'en_US',
+            'permissions' => implode(',', ['read_stream', 'user_photos', 'user_videos', 'manage_pages', 'publish_actions', 'publish_pages']),
+        ];
+
+        $res = $this->testFacebook->post($testUserPath, $params, $this->testFacebook->getApp()->getAccessToken())->getDecodedBody();
+
+        $this->testUserId = $res['id'];
+        $this->testUserToken = $res['access_token'];
+    }
+
+    public function testSuccessUploadVideoByChunk()
+    {
+        $uploader = new FacebookVideoUploader($this->testFacebook);
+
+        $res = $uploader->upload(
+            $filePath = __DIR__ . '/foo.mp4',
```

Comment:
Uploading an actual video to Facebook every time we run unit tests is not ideal. :) These tests should mock the responses.